microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIHealthDynamicBaseTest.java

```diff
@@ -0,0 +1,242 @@
+                .body("status", is("DOWN"),
+                        "checks", hasSize(2),
+                        "checks", containsInAnyOrder(liveCheck, readyCheck));
+
+        setConfigProperties(true, false, true, true);
+        // TODO Java11 readyCheck = Map.of("name", "dummyReadiness", "status", "UP");
+        readyCheck = Collections.unmodifiableMap(new HashMap<String, String>() {
+            {
+                put("name", "dummyReadiness");
+                put("status", "UP");
+            }
+        });
+        get(HealthUrlProvider.healthEndpoint()).then()
+                .statusCode(200)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(2),
+                        "checks", containsInAnyOrder(liveCheck, readyCheck));
+    }
+
+    /**
+     * @tpTestDetails Multiple-component customer scenario - health check status is based on fail-safe CDI bean and MP Config
+     *                property. There is IOException since service is in maintenance but fallback method should be used.
+     * @tpPassCrit Overall and the health check status is down and fallback method is invoked. MP Config change is propagated
+     *             correctly.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void testLivenessEndpointDownToUpInMaintenace() throws Exception {
+        setConfigProperties(false, true, true, true);
+        get(HealthUrlProvider.liveEndpoint()).then()
+                .statusCode(503)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("DOWN"),
+                        "checks", hasSize(1),
+                        "checks.status", hasItems("DOWN"),
+                        "checks.name", containsInAnyOrder("dummyLiveness"));
+
+        setConfigProperties(true, false, true, false);
+        get(HealthUrlProvider.liveEndpoint()).then()
+                .statusCode(200)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(1),
+                        "checks.status", hasItems("UP"),
+                        "checks.name", containsInAnyOrder("dummyLiveness"));
+    }
+
+    /**
+     * @tpTestDetails Multiple-component customer scenario - health check status is based on fail-safe CDI bean and MP Config
+     *                property. There is IOException since service is in maintenance but fallback method should be used.
+     * @tpPassCrit Overall and the health check status is down and fallback method is invoked. MP Config change is propagated
+     *             correctly.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
```

Comment:
Optional: If everything is `@RunAsClient` then just the class could be annotated and not every test.